### PR TITLE
Adding lastSentDate to broken site report 

### PR DESCRIPTION
**Pull request by federicocappelli**
Task/Issue URL: https://app.asana.com/0/1163321984198618/1205860617092609/f

**Description**:

Goal
- See tech design → [✓ Tech design: Report previously submitted site breakage reports](https://app.asana.com/0/481882893211075/1205812132848029/f)
- For each reported website store the first 6 characters of the hash of the domain (not eTLD+1) and last reported date (not timestamp).
- For an existing reported website update the last reported date of the corresponding hash.
- Last reported days are cleared once they're more than 30 days old.

Why Now?
- We have observed a lot of reports that appear to be duplicates
- We implemented this on Android and the Extension
- We found that 10%+ of reports on Android are duplicates, and expect that pattern to hold across products.

**Steps to test this PR**:
- Send a broken website report for website A
- Check the pixel, should not contain lastSentDate
- Send a broken website report for website A
- Check the pixel, should contain lastSentDate with the same day date

Note: 30-days history element expiry can't be easily tested manually, we use unit tests for that.

**Device Testing**:
* [x] iPhone 15

**OS Testing**:

* [x] iOS 17
---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/TabViewController.swift [hunk 1]
```diff
@@ -712,7 +712,7 @@ class TabViewController: UIViewController {
                 controller.popoverPresentationController?.sourceRect = iconView.bounds
             }
             privacyDashboard = controller
-            privacyDashboard?.brokenSiteInfo = getCurrentWebsiteInfo()
+//            privacyDashboard?.breakageAdditionalInfo =
```
Comment (mallexxx) [inline DuckDuckGo/TabViewController.swift]: commented out lines should be removed

File: DuckDuckGo/TabViewController.swift [hunk 2]
```diff
@@ -902,24 +903,49 @@ class TabViewController: UIViewController {
...
+            return nil
+        }
+        return PrivacyDashboardViewController.BreakageAdditionaInfo(currentURL: currentURL,
+                                                                    httpsForced: httpsForced,
+                                                                    ampURLString: linkProtection.lastAMPURLString ?? "",
+                                                                    urlParametersRemoved: linkProtection.urlParametersRemoved,
+                                                                    isDesktop: tabModel.isDesktop)
     }
     
+//    public func getCurrentWebsiteInfo() -> WebsiteBreakage? {
```
Comment (mallexxx) [inline DuckDuckGo/TabViewController.swift]: commented out lines should be removed

**Review by mallexxx — APPROVED** [review]
LGTM! 
Please remove the commented out lines

File: DuckDuckGo/TabViewController.swift [hunk 3]
```diff
@@ -912,22 +912,17 @@ class TabViewController: UIViewController {
...
-                              installedSurrogates: privacyInfo?.trackerInfo.installedSurrogates.map { $0 } ?? [],
-                              isDesktop: tabModel.isDesktop,
-                              tdsETag: ContentBlocking.shared.contentBlockingManager.currentMainRules?.etag ?? "",
-                              ampUrl: linkProtection.lastAMPURLString,
-                              urlParametersRemoved: linkProtection.urlParametersRemoved,
-                              protectionsState: protectionsState)
+        guard let currentURL = url else {
+            return nil
+        }
+        return PrivacyDashboardViewController.BreakageAdditionaInfo(currentURL: currentURL,
```
Comment (jaceklyp) [inline DuckDuckGo/TabViewController.swift]: just noticed a typo, missing L letter here and in the 916 line

Comment (federicocappelli) [reply DuckDuckGo/TabViewController.swift]: well spotted thanks, fixed

**Pull request merged**